server/src/main/java/org/elasticsearch/action/fieldcaps/RequestDispatcher.java

```diff
@@ -108,13 +116,16 @@ void execute() {
             @Override
             public void onFailure(Exception e) {
                 // If we get rejected, mark pending indices as failed and complete
-                final List<String> failedIndices = new ArrayList<>(indexSelectors.keySet());
-                for (String failedIndex : failedIndices) {
-                    final IndexSelector removed = indexSelectors.remove(failedIndex);
-                    assert removed != null;
-                    onIndexFailure.accept(failedIndex, e);
+                try {
+                    final List<String> failedIndices = new ArrayList<>(indexSelectors.keySet());
+                    for (String failedIndex : failedIndices) {
+                        final IndexSelector removed = indexSelectors.remove(failedIndex);
+                        assert removed != null;
+                        onIndexFailure.accept(failedIndex, e);
+                    }
+                } finally {
```

Comment:
Wouldn't it make more sense to quietly record the individual index failure (so essentially catch/suppress inside the loop) instead of propagating a somewhat expected exception further?

Reply:
++, I pushed https://github.com/elastic/elasticsearch/pull/108736/commits/1899f36a967924e6c6de164becba74b2eedd3cc9